### PR TITLE
Make CLI use influxdb for filesystem stats

### LINKED ISSUE #1719: IML CLI fs list / show does not display space, inode usage, or client count.
When running `iml filesystem list` or `iml filesystem show fs` we get something like the following:

```
+------+-----------+-----------------+-----------------+---------+------+------+
| Name | State     | Space           | Inodes          | Clients | MDTs | OSTs |
+------+-----------+-----------------+-----------------+---------+------+------+
| fs   | available | Calculating ... | Calculating ... | 0       | 2    | 10   |
+------+-----------+-----------------+-----------------+---------+------+------+
| fs2  | available | Calculating ... | Calculating ... | 0       | 1    | 10   |
+------+-----------+-----------------+-----------------+---------+------+------+
```

This is because stats have been removed from the main db and are now only available via influx. We should query the relevant /influx endpoint to get these values and fill them in the table.

### PR DESCRIPTION
**Pull request by ip1981**
Factor out influxdb bits into a separate crate.

Closes #1719.

```
[vagrant@adm ~]$ iml filesystem show fs2
+-------------------+----------------------------------------+
| Name              | fs2                                    |
+-------------------+----------------------------------------+
| Space             | 332 MiB / 46 GiB                       |
+-------------------+----------------------------------------+
| Inodes            | 3K / 3M                                |
+-------------------+----------------------------------------+
| State             | available                              |
+-------------------+----------------------------------------+
| Management Server | mds1.local                             |
+-------------------+----------------------------------------+
| MDTs              | fs2-MDT0000                            |
+-------------------+----------------------------------------+
| OSTs              | fs2-OST0004                            |
|                   | fs2-OST0002                            |
|                   | fs2-OST0003                            |
|                   | fs2-OST0001                            |
|                   | fs2-OST0000                            |
|                   | fs2-OST0005                            |
|                   | fs2-OST0008                            |
|                   | fs2-OST0009                            |
|                   | fs2-OST0007                            |
|                   | fs2-OST0006                            |
+-------------------+----------------------------------------+
| Clients           | 1                                      |
+-------------------+----------------------------------------+
| Mount Path        | 10.73.20.11@tcp0:10.73.20.12@tcp0:/fs2 |
+-------------------+----------------------------------------+


[vagrant@adm ~]$ iml filesystem list
+------+-----------+------------------+---------+---------+------+------+
| Name | State     | Space            | Inodes  | Clients | MDTs | OSTs |
+------+-----------+------------------+---------+---------+------+------+
| fs   | available | 333 MiB / 46 GiB | 3K / 3M | 0       | 2    | 10   |
+------+-----------+------------------+---------+---------+------+------+
| fs2  | available | 332 MiB / 46 GiB | 3K / 3M | 1       | 1    | 10   |
+------+-----------+------------------+---------+---------+------+------+
```

Signed-off-by: Igor Pashev <pashev.igor@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/1775)

### CODE REVIEW

File: iml-influx/src/filesystem.rs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+pub struct Response {
+    pub bytes_total: Option<u64>,
+    pub bytes_free: Option<u64>,
+    pub bytes_avail: Option<u64>,
+    pub files_total: Option<u64>,
+    pub files_free: Option<u64>,
+    pub clients: Option<u64>,
+}
+
+impl From<InfluxResponse> for Response {
```
Comment (johnsonw) [inline iml-influx/src/filesystem.rs:26]: Nice :)

File: iml-influx/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[package]
+name = "iml-influx"
+version = "0.1.0"
+authors = ["IML Team <iml@whamcloud.com>"]
+description = "Shared functions and data types related to influx"
+edition = "2018"
+
+[dependencies]
+iml-wire-types = { path = "../iml-wire-types", version = "0.2" }
```
Comment (johnsonw) [inline iml-influx/Cargo.toml:9]: Nit, maybe put version before path for consistency?

File: iml-manager-cli/src/filesystem.rs [hunk 3]
```diff
@@ -109,22 +110,30 @@ async fn detect_filesystem(hosts: Option<String>) -> Result<(), ImlManagerCliErr
...
+                    let s = stats.get(&f.name).cloned().unwrap_or_default();
                     vec![
                         f.label,
                         f.state,
-                        usage(f.bytes_free, f.bytes_total, format_bytes),
-                        usage(f.files_free, f.files_total, format_number),
-                        format_number(f.client_count.unwrap_or(0.0), Some(0)),
+                        usage(s.bytes_free, s.bytes_total, format_bytes),
+                        usage(s.files_free, s.files_total, format_number),
+                        format!("{}", s.clients.unwrap_or(0)),
```
Comment (johnsonw) [inline iml-manager-cli/src/filesystem.rs:136]: Is the format necessary here?

File: iml-manager-cli/src/filesystem.rs [hunk 4]
```diff
@@ -171,7 +187,7 @@ pub async fn filesystem_cli(command: FilesystemCommand) -> Result<(), ImlManager
 
             table.add_row(Row::from(&[
                 "Clients".to_string(),
-                format!("{:0}", fs.client_count.unwrap_or(0.0)),
+                format!("{}", st.clients.unwrap_or(0)),
```
Comment (johnsonw) [inline iml-manager-cli/src/filesystem.rs:190]: Is the format necessary?

File: iml-manager-cli/src/filesystem.rs (hunk 3 above)
Comment (ip1981) [reply iml-manager-cli/src/filesystem.rs:136]: Yes, number to string, I think :)

File: iml-influx/Cargo.toml (hunk 2 above)
Comment (ip1981) [reply iml-influx/Cargo.toml:9]: Maybe in a separate patch. But I'd like something automatic ;-)

**Pull request merged**